packages/realm/bindgen/src/templates/jsi.ts

```diff
@@ -902,6 +912,16 @@ class JsiCppDecls extends CppDecls {
               `,
             )
             .join("\n")}
+
+          // We are returning sentinel values for lists and dictionaries in the
+          // form of Symbol singletons. This is due to not being able to construct
+          // the actual list or dictionary in the current context.
+          case realm::type_List:
+            return ${this.addon.accessCtor("Symbol_for")}.call(_env, "Realm.List");
```

Comment:
`accessCtor` seems ripe for a rename. Perhaps it should be closer linked to "injectable"?